modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.ignite.internal.visor.VisorMultiNodeTask;
+import org.apache.ignite.lang.IgnitePredicate;
+import org.apache.ignite.resources.LoggerResource;
+
+import static org.apache.ignite.events.EventType.EVT_CONSISTENCY_VIOLATION;
+
+/**
+ *
+ */
+public class VisorConsistencyRepairTask extends VisorMultiNodeTask<VisorConsistencyRepairTaskArg, String, String> {
+    /** Serial version uid. */
+    private static final long serialVersionUID = 0L;
+
+    /** Nothing found. */
+    private static final String NOTHING_FOUND = "Consistency violations were not found.";
+
+    /** Found. */
+    public static final String CONSISTENCY_VIOLATIONS_FOUND = "Consistency violations were FOUND";
+
+    /** Violations recorder. */
+    public static final String CONSISTENCY_VIOLATIONS_RECORDED = "Cache consistency violations recorded.";
+
+    /** {@inheritDoc} */
+    @Override protected VisorJob<VisorConsistencyRepairTaskArg, String> job(VisorConsistencyRepairTaskArg arg) {
+        return new VisorConsistencyRepairJob(arg, debug);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected String reduce0(List<ComputeJobResult> results) throws IgniteException {
+        StringBuilder sb = new StringBuilder();
+
+        for (ComputeJobResult res : results) {
+            String data = res.getData();
+
+            if (data != null)
+                sb.append("Node: ").append(res.getNode()).append("\n")
+                    .append("  Result: ").append(data).append("\n");
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     *
+     */
+    private static class VisorConsistencyRepairJob extends VisorJob<VisorConsistencyRepairTaskArg, String> {
+        /** Serial version uid. */
+        private static final long serialVersionUID = 0L;
+
+        /** Injected logger. */
+        @LoggerResource
+        protected IgniteLogger log;
+
+        /** Events. */
+        private final Set<CacheConsistencyViolationEvent> evts = new GridConcurrentHashSet<>();
+
+        /**
+         * @param arg Arguments.
+         * @param debug Debug.
+         */
+        protected VisorConsistencyRepairJob(VisorConsistencyRepairTaskArg arg, boolean debug) {
+            super(arg, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected String run(VisorConsistencyRepairTaskArg arg) throws IgniteException {
+            String cacheName = arg.cacheName();
+            int p = arg.part();
+            int batchSize = 1024;
+
+            GridCacheContext<Object, Object> cctx = ignite.context().cache().cache(cacheName).context();
+
+            if (!cctx.gridEvents().isRecordable(EVT_CONSISTENCY_VIOLATION))
+                throw new UnsupportedOperationException("Consistency violation events recording is disabled on cluster.");
+
+            CacheGroupContext grpCtx = cctx.group();
+
+            GridDhtLocalPartition part = grpCtx.topology().localPartition(p);
+
+            if (part != null) {
```

Comment:
Let's rewrite it with 

```
if (part == null)
    return null;

// Other logic goes here.
```

to reduce indentation level.

Reply:
Fixed